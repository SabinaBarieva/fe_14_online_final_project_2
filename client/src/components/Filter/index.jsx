import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import {
  Button,
  Checkbox,
  Dialog,
  FormControlLabel,
  FormGroup,
  FormLabel,
  Grid,
  IconButton,
  Stack,
  TextField,
  useMediaQuery,
  CircularProgress,
  Typography,
} from '@mui/material';
import CancelIcon from '@mui/icons-material/Cancel';
import FilterAltIcon from '@mui/icons-material/FilterAlt';
import {
  addCategory,
  fetchFilters,
  removeCategory,
  resetFilters,
  setMaxPrice,
  setMinPrice,
} from '../../redux/slices/filtersSlice';
import theme from '../../themes/theme';
import { FilterStyles } from '../../themes/themeFilter';

function Filter() {
  const [filterOpen, setFilterOpen] = useState(false);
  const closeCallback = () => {
    setFilterOpen(false);
  };

  const smallScreen = useMediaQuery(theme.breakpoints.down('md'));

  return !smallScreen ? (
    <FilterSection />
  ) : (
    <>
      <IconButton
        onClick={() => {
          setFilterOpen(true);
        }}>
        <FilterAltIcon />
      </IconButton>

      <Dialog
        fullScreen
        open={filterOpen}
        sx={{
          display: 'flex',
        }}>
        <IconButton onClick={closeCallback}>
          <CancelIcon
            sx={{ position: 'absolute', top: '30px', right: '30px' }}
          />
        </IconButton>

        <FilterSection />
      </Dialog>
    </>
  );
}

function FilterSection() {
  const dispatch = useDispatch();
  const minPrice = useSelector(({ filters }) => filters.minPrice);
  const maxPrice = useSelector(({ filters }) => filters.maxPrice);
  const [cachedMinValue, setCachedMinValue] = useState(minPrice);
  const [cachedMaxValue, setCachedMaxValue] = useState(maxPrice);
  const [priceMinBoundary, setPriceMinBoundary] = useState();
  const [priceMaxBoundary, setPriceMaxBoundary] = useState();

  const { categories, price } = useSelector(
    ({ filters }) => filters.availableFilters
  );
  const selectedCategories = useSelector(({ filters }) => filters.categories);
  const isLoadedFilters = useSelector((state) => state.filters.isLoaded);
  const isLoadingFilters = useSelector((state) => state.filters.isLoading);

  useEffect(() => {
    if (!isLoadedFilters) dispatch(fetchFilters());
    else {
      const { min: priceMin, max: priceMax } = price;
      setPriceMinBoundary(priceMin);
      setPriceMaxBoundary(priceMax);
    }
  }, [isLoadedFilters]);
  const isNumber = (number) =>
    number !== null &&
    number !== undefined &&
    number !== '' &&
    !Number.isNaN(number);
  const categoryCheckboxCallback = ({ target }) => {
    const { checked, name } = target;
    if (checked) dispatch(addCategory(name));
    else dispatch(removeCategory(name));
  };
  const minPriceCallback = ({ target }) => {
    const { value } = target;
    if (isNumber(value) && value < 0) setCachedMinValue(0);
    else if (value === '') setCachedMinValue(null);
    else if (!isNumber(cachedMinValue) && (value === '1' || value === '-1'))
      setCachedMinValue(priceMinBoundary);
    else if (isNumber(value)) setCachedMinValue(value);
  };
  const maxPriceCallback = ({ target }) => {
    const { value } = target;
    if (value < 0) setCachedMaxValue(0);
    else if (value === '') setCachedMaxValue(null);
    else if (!isNumber(cachedMaxValue) && (value === '1' || value === '-1'))
      setCachedMaxValue(priceMaxBoundary);
    else if (isNumber(value)) setCachedMaxValue(value);
  };
  const setPriceCallback = () => {
    dispatch(setMinPrice(cachedMinValue));
    dispatch(setMaxPrice(cachedMaxValue));
  };
  const resetFiltersCallback = () => {
    dispatch(resetFilters());
    setCachedMinValue(null);
    setCachedMaxValue(null);
  };

  if (isLoadingFilters) return <CircularProgress />;
  if (isLoadedFilters)
    return (
      <FilterStyles>
        <Typography
          sx={{ fontSize: '20px', fontWeight: '700', fontFamily: 'Roboto' }}>
          Filter
        </Typography>
        <Stack
          sx={{ width: '300px', fontFamily: 'Roboto' }}
          padding={3}
          spacing={{ xs: 1, sm: 2 }}>
          <FormGroup label="Product Category" sx={{ maxWidth: 300 }}>
            <FormLabel
              sx={{ fontWeight: '550', color: '#000', marginBottom: '10px' }}>
              Product Category
            </FormLabel>
            {[...categories]
              .sort((firstCategory, secondCategory) =>
                firstCategory.name.localeCompare(secondCategory.name)
              )
              .map(({ name: categoryName, id }) => (
                <FormControlLabel
                  key={id}
                  control={<Checkbox />}
                  label={`${categoryName}`}
                  name={id}
                  checked={selectedCategories.includes(id)}
                  labelPlacement="start"
                  onClick={categoryCheckboxCallback}
                  sx={{
                    display: 'flex',
                    justifyContent: 'space-between',
                    margin: '0',
                  }}
                />
              ))}
            <FormLabel
              sx={{
                margin: '10px 0 20px',
                fontWeight: '550',
                color: '#000',
              }}>
              Price range
            </FormLabel>
            <Stack>
              <Grid container justifyContent="space-between">
                <Grid item xs={6}>
                  <TextField
                    size="small"
                    value={isNumber(cachedMinValue) ? cachedMinValue : ''}
                    placeholder={`${priceMinBoundary} $`}
                    type="number"
                    min={priceMinBoundary}
                    onChange={minPriceCallback}
                    onKeyUp={({ key }) => key === 'Enter' && setPriceCallback()}
                    sx={{ width: '95%' }}
                  />
                </Grid>
                <Grid item xs={6} sx={{ textAlign: 'right' }}>
                  <TextField
                    size="small"
                    value={isNumber(cachedMaxValue) ? cachedMaxValue : ''}
                    placeholder={`${priceMaxBoundary} $`}
                    type="number"
                    onChange={maxPriceCallback}
                    onKeyUp={({ key }) => key === 'Enter' && setPriceCallback()}
                    sx={{
                      width: '95%',
                      padding: '0',
                    }}
                  />
                </Grid>
              </Grid>
              <Grid container>
                <Grid item xs={12}>
                  <Button
                    sx={{
                      minWidth: '100%',
                      borderRadius: '8px',
                      marginTop: '20px',
                      textTransform: 'capitalize',
                      background: '#000',
                    }}
                    variant="contained"
                    onClick={setPriceCallback}>
                    Set Price
                  </Button>
                </Grid>
                <Grid item xs={6} md={12}>
                  <Button
                    sx={{
                      width: { xs: '98%', md: '100%' },
                      marginTop: { xs: '40%', md: '65px' },
                      border: '1px solid #000',
                      borderRadius: '8px',
                      textTransform: 'capitalize',
                    }}
                    variant="outlined"
                    onClick={resetFiltersCallback}>
                    Clear Filter
                  </Button>
                </Grid>
                <Grid item xs={6} sx={{ textAlign: 'right' }}>
                  <Button
                    sx={{
                      display: { md: 'none' },
                      width: { xs: '98%' },
                      marginTop: { xs: '40%' },
                      borderRadius: '8px',
                      textAlign: 'right',
                      textTransform: 'capitalize',
                      background: '#000',
                    }}
                    variant="contained"
                    onClick={resetFiltersCallback}>
                    Apply
                  </Button>
                </Grid>
              </Grid>
            </Stack>
          </FormGroup>
        </Stack>
      </FilterStyles>
    );
}
export default Filter;
