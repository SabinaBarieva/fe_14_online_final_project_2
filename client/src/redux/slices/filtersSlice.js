import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import getProductsFilters from '../../api/getProductsFilters';
import { setErrorMessage } from './errorsSlice';

const initialState = {
  categories: [],
  minPrice: null,
  maxPrice: null,
  availableFilters: {},
  isLoading: false,
  isLoaded: false,
};

export const fetchFilters = createAsyncThunk(
  'filters/fetch',
  async (_, { dispatch }) => {
    try {
      const filters = await getProductsFilters();
      return filters;
    } catch (error) {
      dispatch(setErrorMessage({ error: error.message }));
      throw error;
    }
  }
);
const filtersSlice = createSlice({
  name: 'filters',
  initialState,
  reducers: {
    addCategory: ({ categories }, { payload: { category } }) => {
      if (!categories.includes(category)) categories.push(category);
    },
    removeCategory: ({ categories }, { payload: { category } }) => {
      const categoryIndex = categories.findIndex(
        (arrayItem) => arrayItem === category
      );
      if (categoryIndex > -1) categories.splice(categoryIndex, 1);
    },

    setMinPrice: (state, { payload: { price } }) => {
      state.minPrice = price;
    },
    setMaxPrice: (state, { payload: { price } }) => {
      state.maxPrice = price;
    },
    resetFilters: (state) => ({
      ...initialState,
      availableFilters: { ...state.availableFilters },
      isLoaded: state.isLoaded,
      isLoading: state.isLoading,
    }),
  },
  extraReducers: (builder) => {
    builder.addCase(fetchFilters.pending, (state) => {
      state.isLoading = true;
    });
    builder.addCase(fetchFilters.fulfilled, (state, action) => {
      state.isLoading = false;
      state.isLoaded = true;
      state.availableFilters = action.payload;
    });
    builder.addCase(fetchFilters.rejected, (state) => {
      state.isLoading = false;
      state.isLoaded = false;
    });
  },
});

const {
  addCategory: addCategoryAction,
  removeCategory: removeCategoryAction,
  setMinPrice: setMinPriceAction,
  setMaxPrice: setMaxPriceAction,
} = filtersSlice.actions;
export const addCategory = (category) => addCategoryAction({ category });
export const removeCategory = (category) => removeCategoryAction({ category });
export const setMinPrice = (price) => setMinPriceAction({ price });
export const setMaxPrice = (price) => setMaxPriceAction({ price });
export const { resetFilters } = filtersSlice.actions;

export default filtersSlice.reducer;
