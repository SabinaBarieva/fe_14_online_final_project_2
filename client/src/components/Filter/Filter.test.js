import React from 'react';
import { Provider } from 'react-redux';
import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import store from '../../redux/store';
import Filter from '.';
import { fetchFilters, resetFilters } from '../../redux/slices/filtersSlice';

jest.setTimeout(10000);
const minPriceBoundary = 9;
const maxPriceBoundary = 9876;
const filterCategories = {
  categories: [
    { id: 'test-category-1', name: 'Test Category 1', quantity: 7 },
    { id: 'test-category-2', name: 'Test Category 2', quantity: 7 },
    { id: 'test-category-3', name: 'Test Category 3', quantity: 8 },
    { id: 'test-category-4', name: 'Test Category 4', quantity: 5 },
    { id: 'test-category-5', name: 'Test Category 5', quantity: 10 },
    { id: 'test-category-6', name: 'Test Category 6', quantity: 5 },
  ],
  price: { min: minPriceBoundary, max: maxPriceBoundary },
};

const server = setupServer(
  rest.get('/api/product-filters', (req, res, ctx) =>
    res(ctx.json(filterCategories))
  )
);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
const minPriceInput = () => screen.getByLabelText('Min');
const maxPriceInput = () => screen.getByLabelText('Max');
const categoriesCheckboxes = () => screen.getAllByLabelText(/Test Category.*/);
const resetButton = () => screen.getByText('Clear Filter');

const init = async () => {
  store.dispatch(fetchFilters());
  store.dispatch(resetFilters());
  render(<Component />);
  await screen.findByText('Filter');
};

function Component() {
  return (
    <Provider store={store}>
      <Filter
        priceMinBoundary={minPriceBoundary}
        priceMaxBoundary={maxPriceBoundary}
      />
    </Provider>
  );
}
describe('Filter component and filterSlice', () => {
  test.skip('snapshot renders correctly', async () => {
    await init();
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Component renders', async () => {
    await init();
    expect(screen.getByText('Filter')).toBeInTheDocument();
    expect(screen.getByText('Product Category')).toBeInTheDocument();
    expect(screen.getByText('Price range')).toBeInTheDocument();
    expect(categoriesCheckboxes().length).toEqual(6);
  });

  test('Set category', async () => {
    await init();
    const checkbox = categoriesCheckboxes()[0];
    expect(checkbox.name).toEqual('test-category-1');
    expect(checkbox.checked).toEqual(false);
    act(() => {
      userEvent.click(checkbox);
    });

    expect(checkbox.checked).toEqual(true);
    act(() => {
      userEvent.click(checkbox);
    });
    expect(checkbox.checked).toEqual(false);
  });

  test('Set price', async () => {
    await init();
    act(() => {
      userEvent.type(minPriceInput(), String(minPriceBoundary + 10));
    });
    expect(minPriceInput().value).toEqual(String(minPriceBoundary + 10));
    act(() => {
      userEvent.type(maxPriceInput(), String(maxPriceBoundary - 10));
    });
    expect(maxPriceInput().value).toEqual(String(maxPriceBoundary - 10));
  });

  test('Reset filter', async () => {
    await init();
    act(() => {
      categoriesCheckboxes().forEach((checkbox) => {
        userEvent.click(checkbox);
        expect(checkbox.checked).toEqual(true);
      });
    });
    act(() => {
      userEvent.type(minPriceInput(), String(minPriceBoundary + 10));
    });
    expect(minPriceInput().value).toEqual(String(minPriceBoundary + 10));
    act(() => {
      userEvent.type(maxPriceInput(), String(maxPriceBoundary - 10));
    });
    expect(maxPriceInput().value).toEqual(String(maxPriceBoundary - 10));
    act(() => {
      userEvent.click(resetButton());
    });
    categoriesCheckboxes().forEach((checkbox) =>
      expect(checkbox.checked).toEqual(false)
    );
    expect(minPriceInput().value).toEqual('');
    expect(maxPriceInput().value).toEqual('');
  });

  test('Set min and max price out of boundaries', async () => {
    await init();
    act(() => {
      userEvent.type(minPriceInput(), String(minPriceBoundary - 1));
      userEvent.type(maxPriceInput(), String(maxPriceBoundary + 1));
      userEvent.click(minPriceInput());
    });

    expect(minPriceInput().value).toEqual(String(minPriceBoundary));
    expect(maxPriceInput().value).toEqual(String(maxPriceBoundary));
    act(() => {
      userEvent.click(resetButton());
    });
    act(() => {
      userEvent.type(minPriceInput(), String(minPriceBoundary));
      userEvent.type(maxPriceInput(), String(maxPriceBoundary));
      userEvent.click(minPriceInput());
    });

    expect(minPriceInput().value).toEqual(String(minPriceBoundary));
    expect(maxPriceInput().value).toEqual(String(maxPriceBoundary));
    act(() => {
      userEvent.click(resetButton());
    });
    act(() => {
      userEvent.type(minPriceInput(), String(minPriceBoundary + 1));
      userEvent.type(maxPriceInput(), String(maxPriceBoundary - 1));
      userEvent.click(minPriceInput());
    });
    expect(minPriceInput().value).toEqual(String(minPriceBoundary + 1));
    expect(maxPriceInput().value).toEqual(String(maxPriceBoundary - 1));
  });
});
