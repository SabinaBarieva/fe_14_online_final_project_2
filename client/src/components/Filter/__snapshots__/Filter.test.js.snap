// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component and filterSlice snapshot renders correctly 1`] = `
<aside
  className="css-1kltobf"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-rrlk9y-MuiTypography-root"
  >
    Filter
  </p>
  <div
    className="MuiStack-root css-1ec0wt2-MuiStack-root"
  >
    <div
      className="MuiFormGroup-root css-19tqrsw-MuiFormGroup-root"
      label="Product Category"
    >
      <label
        className="MuiFormLabel-root MuiFormLabel-colorPrimary css-13aoscy-MuiFormLabel-root"
      >
        Product Category
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1ilel39-MuiFormControlLabel-root"
        onClick={[Function]}
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            name="test-category-1"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Test Category 1
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1ilel39-MuiFormControlLabel-root"
        onClick={[Function]}
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            name="test-category-2"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Test Category 2
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1ilel39-MuiFormControlLabel-root"
        onClick={[Function]}
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            name="test-category-3"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Test Category 3
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1ilel39-MuiFormControlLabel-root"
        onClick={[Function]}
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            name="test-category-4"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Test Category 4
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1ilel39-MuiFormControlLabel-root"
        onClick={[Function]}
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            name="test-category-5"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Test Category 5
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1ilel39-MuiFormControlLabel-root"
        onClick={[Function]}
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            name="test-category-6"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Test Category 6
        </span>
      </label>
      <label
        className="MuiFormLabel-root MuiFormLabel-colorPrimary css-16q5tl2-MuiFormLabel-root"
      >
        Price range
      </label>
      <div
        className="MuiStack-root css-nen11g-MuiStack-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root css-19gy475-MuiFormControl-root-MuiTextField-root"
              min={9}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor=":r0:"
                id=":r0:-label"
              >
                Min
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r0:"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="tel"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      Min
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1gejpqg-MuiGrid-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root css-115svll-MuiFormControl-root-MuiTextField-root"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor=":r1:"
                id=":r1:-label"
              >
                Max
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r1:"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="tel"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      Max
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container css-drc2vn-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1j60t1q-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Clear Filter
            </button>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1bufe5z-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Set Price
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</aside>
`;
